cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -0,0 +1,165 @@
+		return errors.Wrap(ErrCannotUpdateExistingQueue, errHelp)
+	}
+
+	// workerQueueToCreate does not exist so we add it
+	wQueueListToCreate = append(requestedDeployment.WorkerQueues, *workerQueueToCreate) //nolint
+
+	// update the deployment with the new list of worker-queues
+	err = deployment.Update(requestedDeployment.ID, "", ws, "", 0, 0, 0, wQueueListToCreate, true, client)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "worker-queue %s for %s in %s workspace created\n", workerQueueToCreate.Name, requestedDeployment.ID, ws)
+	return nil
+}
+
+// setWorkerQueueValues sets values for MinWorkerCount, MaxWorkerCount and WorkerConcurrency
+// Default values are used if the user did not request any
+
+func setWorkerQueueValues(wQueueMin, wQueueMax, wQueueConcurrency int, workerQueueToCreate *astro.WorkerQueue, workerQueueDefaultOptions astro.WorkerQueueDefaultOptions) *astro.WorkerQueue {
+	if wQueueMin != 0 {
+		// use the value from the user input
+		workerQueueToCreate.MinWorkerCount = wQueueMin
+	} else {
+		// set default value as user input did not have it
+		workerQueueToCreate.MinWorkerCount = workerQueueDefaultOptions.MinWorkerCount.Default
+	}
+
+	if wQueueMax != 0 {
+		// use the value from the user input
+		workerQueueToCreate.MaxWorkerCount = wQueueMax
+	} else {
+		// set default value as user input did not have it
+		workerQueueToCreate.MaxWorkerCount = workerQueueDefaultOptions.MaxWorkerCount.Default
+	}
+
+	if wQueueConcurrency != 0 {
+		workerQueueToCreate.WorkerConcurrency = wQueueConcurrency
+	} else {
+		workerQueueToCreate.WorkerConcurrency = workerQueueDefaultOptions.WorkerConcurrency.Default
+	}
+	return workerQueueToCreate
+}
+
+// GetWorkerQueueDefaultOptions calls the workerqueues query
+// It returns WorkerQueueDefaultOptions if the query succeeds
+// An error is returned if it fails
+
+func GetWorkerQueueDefaultOptions(client astro.Client) (astro.WorkerQueueDefaultOptions, error) {
+	var (
+		workerQueueDefaultOptions astro.WorkerQueueDefaultOptions
+		err                       error
+	)
+	workerQueueDefaultOptions, err = client.GetWorkerQueueOptions()
+	if err != nil {
+		return astro.WorkerQueueDefaultOptions{}, err
+	}
+	return workerQueueDefaultOptions, nil
+}
+
+// IsWorkerQueueInputValid checks if the requestedWorkerQueue adheres to the floor and ceiling set in the defaultOptions
+// if it adheres to them, it returns nil
+// ErrInvalidWorkerQueueOption is returned if min, max or concurrency are out of range
+
+func IsWorkerQueueInputValid(requestedWorkerQueue *astro.WorkerQueue, defaultOptions astro.WorkerQueueDefaultOptions) error {
+	var errorMessage string
+	if !(requestedWorkerQueue.MinWorkerCount >= defaultOptions.MinWorkerCount.Floor) ||
+		!(requestedWorkerQueue.MinWorkerCount <= defaultOptions.MinWorkerCount.Ceiling) {
+		errorMessage = fmt.Sprintf("min worker count must be between %d and %d", defaultOptions.MinWorkerCount.Floor, defaultOptions.MinWorkerCount.Ceiling)
```

Comment:
We should probably prefix these errors with - `Failed to create worker queue` so that users know which part of their input is malformatted / invalid

Reply:
@frank-ye The error is wrapped as below for any invalid min, max or concurrency option that a user requests:

```
min worker count must be between 1 and 20: worker queue option is invalid
```
The range above (1 - 20) is fake from a test. I can walk you through the real interaction if you'd like to see if we can improve the user facing error message. The reason I stayed away from adding a verb like `create` in the error output is so that we can re-use the same method regardless of a user trying to create or update a worker queue.

Reply:
No need your justification is very clear!